command without starting `\` defined by `\newcommand` cause dynamically built snippet not generated

## Preliminary questions [Required]
If using `\newcommand` to define a command that is not starting with `\`(by changing catcode), errror `Unaccepted argument type` will occur, and the following function will not be usable:
>The files of a LaTeX project are searched for any already used commands in the form `mycommand` followed by several `{}` groups. Then, a snippet is dynamically built for each of them and they are added to the command completion list.

### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes


## Describe the bug [Required]

A clear and concise description of what the bug is.

### To Reproduce

Steps to reproduce the behavior:
1. Past the following MME to the editor, then save
```latex
\documentclass{article}
\catcode`\。=\active
\newcommand{。}{．} %<----unusual usage

\newcommand{\degree}{^\circ}

\begin{document}

\end{document}
```

2. type "\deg", `\degree` is not listed in the auto-completion list
![image](https://user-images.githubusercontent.com/8469304/205085193-014ebd2e-def8-4374-a72a-0154ea2e14b3.png)

3. comment the line and save, the dynamically built snippet generated properly
```latex
\documentclass{article}
\catcode`\。=\active
%\newcommand{。}{．} %<----unusual usage

\newcommand{\degree}{^\circ}

\begin{document}

\end{document}
```
![image](https://user-images.githubusercontent.com/8469304/205086430-c16b7fb2-cbb9-4c30-ac3e-2595b278861f.png)

### Expected behavior

Such usage should not be treated as error and the dynamically built snippet should be generated properly




## Logs [Required]

Please paste the whole log messages here, not parts of ones. The log should start with `Initializing LaTeX Workshop`. It is very important to identify problems.

### LaTeX Workshop Output [Required]

```
To access the log, click the 'TeX' icon on the Activity Bar on the left side, select 'View Log Messages', then select 'View LaTeX Workshop extension log'.

[23:13:55] Initializing LaTeX Workshop.
[23:13:55] Extension root: c:\Users\Terrans\.vscode\extensions\james-yu.latex-workshop-9.0.1
[23:13:55] $PATH: C:\Program Files (x86)\VMware\VMware Workstation\bin\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\System32;C:\Windows;C:\Windows\System32\wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files (x86)\GnuPG\bin;C:\Program Files\Git\cmd;C:\Windows\System32;C:\Windows;C:\Windows\System32\wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Wolfram Research\WolframScript\;C:\texlive\2022\bin\win32;C:\Program Files\dotnet\;C:\Users\Terrans\AppData\Local\Microsoft\WindowsApps;C:\Users\Terrans\AppData\Local\Microsoft\WindowsApps;D:\Tools\android-adb\platform-tools;C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.2.2\bin;;C:\Users\Terrans\Documents\aria2;C:\Users\Terrans\Documents\ffmpeg\bin;
[23:13:55] $SHELL: undefined
[23:13:55] $LANG: undefined
[23:13:55] $LC_ALL: undefined
[23:13:55] process.platform: win32
[23:13:55] process.arch: x64
[23:13:55] vscode.env.appName: Visual Studio Code
[23:13:55] vscode.env.remoteName: undefined
[23:13:55] vscode.env.uiKind: 1
[23:13:55] Configuration for workspace: undefined
[23:13:55] editor.acceptSuggestionOnEnter: "on"
[23:13:55] latex-workshop.bind.enter.key: true
[23:13:55] latex-workshop.docker.enabled: false
[23:13:55] latex-workshop.docker.image.latex: ""
[23:13:55] latex-workshop.hover.preview.mathjax.extensions: []
[23:13:55] latex-workshop.intellisense.package.enabled: true
[23:13:55] latex-workshop.intellisense.update.aggressive.enabled: false
[23:13:55] latex-workshop.intellisense.update.delay: 1000
[23:13:55] latex-workshop.latex.autoBuild.run: "never"
[23:13:55] latex-workshop.latex.build.forceRecipeUsage: true
[23:13:55] latex-workshop.latex.outDir: "%DIR%"
[23:13:55] latex-workshop.latex.recipes: [
 {
  "name": "xetex",
  "tools": [
   "xetex"
  ]
 },
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "latexmk (xelatex)",
  "tools": [
   "xelatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[23:13:55] latex-workshop.latex.tools: [
 {
  "name": "xetex",
  "command": "xetex",
  "args": [
   "--synctex=-1",
   "--interaction=nonstopmode"
  ],
  "env": {}
 },
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "xelatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-xelatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texmintex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
]
[23:13:55] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[23:13:55] Creating a new file watcher.
[23:13:55] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[23:13:55] Creating PDF file watcher.
[23:13:55] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[23:13:55] Creating Bib file watcher.
[23:13:55] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[23:13:55] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[23:13:56] [Server] Creating LaTeX Workshop http and websocket server.
[23:13:56] LaTeX Workshop initialized.
[23:13:56] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[23:13:56] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[23:13:56] Current workspace folders: undefined
[23:13:56] Found root file from active editor: d:\TeXTemp\测试最小实例.tex
[23:13:56] Root file changed: from undefined to d:\TeXTemp\测试最小实例.tex
[23:13:56] Start to find all dependencies.
[23:13:56] Root file languageId: latex
[23:13:56] Reset file watcher.
[23:13:56] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:13:56] Parse fls file.
[23:13:56] Cannot find fls file: d:\TeXTemp\测试最小实例.fls
[23:13:56] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":62898}
[23:13:56] Added to file watcher: d:\TeXTemp\测试最小实例.tex
[23:13:56] [Server] valdOrigin is http://127.0.0.1:62898
[23:13:56] Snippet data loaded.
[23:13:56] Checking for duplicate labels: d:\TeXTemp\测试最小实例.tex.
[23:13:58] Manager.fileWatcher.getWatched: {"d:\\TeXTemp":["测试最小实例.tex"]}
[23:13:58] Manager.filesWatched: ["d:\\TeXTemp\\测试最小实例.tex"]
[23:13:58] BibWatcher.bibWatcher.getWatched: {}
[23:13:58] BibWatcher.bibsWatched: []
[23:13:58] PdfWatcher.pdfWatcher.getWatched: {}
[23:13:58] PdfWatcher.pdfsWatched: []
[23:13:58] PdfWatcher.watchedPdfVirtualUris: []
[23:13:58] PdfWatcher.ignoredPdfUris: []
[23:14:11] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:11] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:14:11] onDidSaveTextDocument triggered: file:///d:/TeXTemp/测试最小实例.tex
[23:14:11] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:11] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:14:31] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:31] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:14:31] Checking for duplicate labels: d:\TeXTemp\测试最小实例.tex.
[23:14:31] onDidSaveTextDocument triggered: file:///d:/TeXTemp/测试最小实例.tex
[23:14:31] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:31] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:14:31] Checking for duplicate labels: d:\TeXTemp\测试最小实例.tex.
[23:14:50] onDidSaveTextDocument triggered: file:///d:/TeXTemp/测试最小实例.tex
[23:14:50] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:50] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
[23:14:57] onDidSaveTextDocument triggered: file:///d:/TeXTemp/测试最小实例.tex
[23:14:57] File watcher - file changed: d:\TeXTemp\测试最小实例.tex
[23:14:57] Parsing a file and its subfiles: d:\TeXTemp\测试最小实例.tex
```

### Developer Tools Console [Required]

```
To access the log, click 'help' -> 'Toggle Developer Tools' -> 'Console'. Paste anything suspicious.


workbench.desktop.main.js:sourcemap:2302 [[object Object]]Unaccepted argument type
$onExtensionRuntimeError @ workbench.desktop.main.js:sourcemap:2302
workbench.desktop.main.js:sourcemap:2302 Error: Unaccepted argument type
	at CommandNameDuplicationDetector.has (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:319:19)
	at CommandFinder.getCmdFromNode (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:110:53)
	at c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:67:37
	at Array.forEach (<anonymous>)
	at CommandFinder.getCmdFromNodeArray (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:66:15)
	at Command.update (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\command.js:229:56)
	at Manager.updateCompleter (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\components\manager.js:856:46)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Manager.updateCompleterOnChange (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\components\manager.js:837:9)

[[object Object]]Unaccepted argument type
$onExtensionRuntimeError @ mainThreadExtensionService.ts:79
mainThreadExtensionService.ts:80 Error: Unaccepted argument type
	at CommandNameDuplicationDetector.has (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:319:19)
	at CommandFinder.getCmdFromNode (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:110:53)
	at c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:67:37
	at Array.forEach (<anonymous>)
	at CommandFinder.getCmdFromNodeArray (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\commandlib\commandfinder.js:66:15)
	at Command.update (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\providers\completer\command.js:229:56)
	at Manager.updateCompleter (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\components\manager.js:856:46)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Manager.updateCompleterOnChange (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\components\manager.js:837:9)
	at async Manager.onWatchedFileChanged (c:\Users\****\.vscode\extensions\james-yu.latex-workshop-8.28.0\out\src\components\manager.js:778:13)
```



## Screenshots

If applicable, add screenshots to help explain your problem.




## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: [e.g. Windows 11]
 - VS Code version: [e.g. 1.73.1]
 - Extension version: [e.g. 9.0.1]
 - TeX distribution version: [e.g. TeX Live 2022]




## Additional questions

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

### Additional context

This issue start appear from version 8.28.0, no such issue before this version

--- a/src/providers/completer/commandlib/commandfinder.ts
+++ b/src/providers/completer/commandlib/commandfinder.ts
@@ -85,8 +85,11 @@ export class CommandFinder {
                 cmds.push(cmd)
                 commandNameDuplicationDetector.add(node.name)
             }
-            if (['newcommand', 'renewcommand', 'providecommand', 'DeclareMathOperator', 'DeclarePairedDelimiter', 'DeclarePairedDelimiterX', 'DeclarePairedDelimiterXPP'].includes(node.name.replace(/\*$/, '')) &&
-                Array.isArray(node.args) && node.args.length > 0) {
+            const newCommandDeclarations = ['newcommand', 'renewcommand', 'providecommand', 'DeclareMathOperator', 'DeclarePairedDelimiter', 'DeclarePairedDelimiterX', 'DeclarePairedDelimiterXPP']
+            if (newCommandDeclarations.includes(node.name.replace(/\*$/, '')) &&
+                Array.isArray(node.args) && node.args.length > 0 &&
+                latexParser.isGroup(node.args[0]) && node.args[0].content.length > 0 &&
+                latexParser.isCommand(node.args[0].content[0])) {
                 const label = (node.args[0].content[0] as latexParser.Command).name
                 let tabStops = ''
                 let args = ''

